numbat/src/ffi/lists.rs

```diff
@@ -11,10 +11,12 @@ pub fn len(mut args: Args) -> Result<Value> {
 }
 
 pub fn head(mut args: Args) -> Result<Value> {
-    let mut list = list_arg!(args);
+    let list = list_arg!(args);
 
-    if let Some(first) = list.pop_front() {
-        Ok(first)
+    // We don't need to pop or drop anything, the whole allocation will
+    // be dropped if we're its last owner.
+    if let Some(first) = list.front() {
+        Ok(first.clone())
```

Comment:
If the `first` element is a large object itself (a struct or a list), we now need an additional clone here?

Reply:
Huum, most of the time it's the same as before since we were already cloning the whole list.
But now it's true that we could check if we're the only owner of the list and only clone it when someone else is using the list.

Reply:
Fixed in: https://github.com/sharkdp/numbat/pull/515/commits/f15ea87ef8c890ccbbd20ed22416638f16c2982e